src/silx/resources/__init__.py

```diff
@@ -217,14 +203,14 @@ def _get_package_and_resource(resource, default_directory=None):
     else:
         prefix = "silx"
         if default_directory is not None:
-            resource = os.path.join(default_directory, resource)
+            resource = f"{default_directory}/{resource}"
```

Comment:
this one I don't get

Reply:
Having a 2nd look at this, I think we can remove the `default_directory` stuff, it's only accessible from private API and shouldn't be needed.

Reply:
Forget it, this is need to simplify loading icons without passing `gui/icons/` as a prefix